src/System Application/App/AppSource Gallery/src/AppSourceProductDetails.Page.al

```diff
@@ -0,0 +1,395 @@
+                field(Offer_ProductType; AppSourceProductManager.GetStringValue(ProductObject, 'productType'))
+                {
+                    Caption = 'Product Type';
+                    ToolTip = 'Specifies the Product Type';
+                    Visible = false;
+                }
+                field(Offer_DisplayName; AppSourceProductManager.GetStringValue(ProductObject, 'displayName'))
+                {
+                    Caption = 'Display Name';
+                    ToolTip = 'Specifies the Display Name';
+                }
+                field(Offer_PublisherID; AppSourceProductManager.GetStringValue(ProductObject, 'publisherId'))
+                {
+                    Caption = 'Publisher ID';
+                    ToolTip = 'Specifies the Publisher ID';
+                    Visible = false;
+                }
+                field(Offer_PublisherDisplayName; AppSourceProductManager.GetStringValue(ProductObject, 'publisherDisplayName'))
+                {
+                    Caption = 'Publisher Display Name';
+                    ToolTip = 'Specifies the Publisher Display Name';
+                }
+                field(Offer_PublisherType; AppSourceProductManager.GetStringValue(ProductObject, 'publisherType'))
+                {
+                    Caption = 'Publisher Type';
+                    ToolTip = 'Specifies the Publisher Type';
+                }
+                field(Offer_LastModifiedDateTime; AppSourceProductManager.GetStringValue(ProductObject, 'lastModifiedDateTime'))
+                {
+                    Caption = 'Last Modified Date Time';
+                    ToolTip = 'Specifies the Last Modified Date Time';
+                }
+            }
+            group(DescriptionGroup)
+            {
+                ShowCaption = false;
+
+                field(Description_Description; AppSourceProductManager.GetStringValue(ProductObject, 'description'))
+                {
+                    Caption = 'Description';
+                    MultiLine = true;
+                    ExtendedDatatype = RichContent;
+                    ApplicationArea = All;
+                    ToolTip = 'Specifies the Description';
+                }
+            }
+            group(PlansGroup)
+            {
+                Caption = 'Plans';
+                Visible = PlansAreVisible;
+
+                field("PlansOverview"; PlansOverview)
+                {
+                    ExtendedDatatype = RichContent;
+                    MultiLine = true;
+                    Caption = 'Plans Overview';
+                    ToolTip = 'Specifies the overview of all plans';
+                }
+            }
+
+
+            group(RatingGroup)
+            {
+                Caption = 'Rating';
+
+                field(Rating_Popularity; AppSourceProductManager.GetStringValue(ProductObject, 'popularity'))
+                {
+                    Caption = 'Popularity';
+                    ToolTip = 'Specifies the Popularity';
```

Comment:
"Specifies the number of downloads" or "Specifies the popularity on a scale of 1-5" or something explaining how popularity is measured

Reply:
To address in this PR